kubernetes-examples/src/main/java/io/fabric8/kubernetes/examples/ExecLoopExample.java

```diff
@@ -45,22 +48,34 @@ public static void main(String[] args) throws InterruptedException, IOException
       podName = args[0];
     }
 
-    Config config = new ConfigBuilder().withMasterUrl(master).build();
+      Config config = new ConfigBuilder()
+        .withMasterUrl(master)
+        .build();
 
+      ScheduledExecutorService executorService = Executors.newScheduledThreadPool(20);
       try (OpenShiftClient client = new DefaultOpenShiftClient(config)) {
         for (int i = 0; i < 10; System.out.println("i=" + i), i++) {
-          ExecutorService executorService = Executors.newSingleThreadExecutor();
-          try (ExecWatch watch = client.pods().withName(podName).redirectingOutput().exec("date");
-               InputStreamPumper pump = new InputStreamPumper(watch.getOutput(), new SystemOutCallback())) {
-              executorService.submit(pump);
-              Thread.sleep(2000);
+          ExecWatch watch = null;
+          InputStreamPumper pump = null;
+          try {
+               watch = client.pods().withName(podName).redirectingOutput().exec("date");
+               pump = new InputStreamPumper(watch.getOutput(), new SystemOutCallback());
+               executorService.submit(pump);
+               Future<String> outPumpFuture = executorService.submit(pump, "Done");
+               executorService.scheduleAtFixedRate(new FutureChecker("Pump " + (i + 1), outPumpFuture), 0, 2, TimeUnit.SECONDS);
+
+               //We need to wait or the pumper (ws -> System.out) will not be able to print the message.
+               Thread.sleep(1000);
```

Comment:
Shouldn't the pumper finish when it has pumped all of the input? Is there a way to restart writing output to the `out` pipe after the websocket is closed? I see your point, and realize I'm probably missing a use case somewhere here.